src/core/server/http/security_response_headers_config.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+const strictTransportSecuritySchema = schema.object({
+  // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+  maxAge: schema.duration({ defaultValue: '1Y' }),
+  includeSubDomains: schema.boolean({ defaultValue: false }),
+  preload: schema.boolean({ defaultValue: false }),
+});
+
+export const securityResponseHeadersSchema = schema.object({
+  strictTransportSecurity: schema.oneOf([strictTransportSecuritySchema, schema.literal(null)], {
+    defaultValue: strictTransportSecuritySchema.validate({}),
+  }),
+  xContentTypeOptions: schema.oneOf([schema.literal('nosniff'), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
+  referrerPolicy: schema.oneOf(
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
+    [
+      schema.literal('no-referrer'),
+      schema.literal('no-referrer-when-downgrade'),
+      schema.literal('origin'),
+      schema.literal('origin-when-cross-origin'),
+      schema.literal('same-origin'),
+      schema.literal('strict-origin'),
+      schema.literal('strict-origin-when-cross-origin'),
+      schema.literal('unsafe-url'),
+      schema.literal(null),
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
+    defaultValue: 'camera=(), microphone=()', // disables camera and microphone access in Kibana and any embedded pages
+  }),
+  disableEmbedding: schema.boolean({ defaultValue: false }), // is used to control X-Frame-Options and CSP headers
```

Comment:
When enabled, this changes the default `Content-Security-Policy` header (by adding the `frame-ancestors` directive) and adds the `X-Frame-Options` header.

Even though it appears that [all of Kibana's target browsers support `frame-ancestors`](https://caniuse.com/mdn-http_headers_csp_content-security-policy_frame-ancestors), we still include `X-Frame-Options` for a couple of reasons:
1. If the user has customized the `Content-Security-Policy` for some reason, we won't add the `frame-ancestors` directive -- so including `X-Frame-Options` as a backup is a good idea
2. Naive security scanning tools can see a missing `X-Frame-Options` as a problem, even if there is a `Content-Security-Policy` present with a `frame-ancestors` directive

In the future (8.0 release?) I would love to change this to be enabled by default.